Signal/src/environment/SignalApp.m

```diff
@@ -152,24 +152,24 @@ - (AccountManager *)accountManager
 
 #pragma mark - View Convenience Methods
 
-- (void)presentConversationForRecipientId:(NSString *)recipientId
+- (void)presentConversationForRecipientId:(NSString *)recipientId animated:(BOOL)isAnimated
 {
-    [self presentConversationForRecipientId:recipientId action:ConversationViewActionNone];
+    [self presentConversationForRecipientId:recipientId action:ConversationViewActionNone animated:(BOOL)isAnimated];
 }
 
-- (void)presentConversationForRecipientId:(NSString *)recipientId action:(ConversationViewAction)action
+- (void)presentConversationForRecipientId:(NSString *)recipientId
+                                   action:(ConversationViewAction)action
+                                 animated:(BOOL)isAnimated
 {
-    DispatchMainThreadSafe(^{
```

Comment:
There's already a `DispatchMainThreadSafe` check farther down the stack frame before manipulating UIKit. The OWSPrimaryStorage stuff can happen off the main thread (though in practice these methods all already assert is main thread, down the stack frame)